### PR TITLE
artifact update api integration

### PR DESCRIPTION
**Pull request by bboozzoo**
https://tracker.mender.io/browse/MEN-737

@mendersoftware/rndity @pasinskim @maciejmrowiec

### CODE REVIEW

File: docs/devices_api.yml [hunk 1]
```diff
@@ -180,36 +183,47 @@ definitions:
...
-            type: string
-            format: date-time
-          yocto_id:
+          source:
+            type: object
+            properties:
+              uri:
+                type: string
+                format: url
+                description: URL to fetch the image from
```
Comment (pasinskim) [inline docs/devices_api.yml]: s/image/artifact/

File: docs/devices_api.yml [hunk 2]
```diff
@@ -180,36 +183,47 @@ definitions:
...
       application/json:
-        image:
-          uri: 'https://aws.my_update_bucket.com/yocto_image123'
-          checksum: cc436f982bc60a8255fe1926a450db5f195a19ad
-          id: f81d4fae-7dec-11d0-a765-00a0c91e6bf6
-          expire: 2016-03-11T13:03:17.063493443Z
-          yocto_id: core-image-full-cmdline-20160330201408
         id: w81s4fae-7dec-11d0-a765-00a0c91e6bf6
+        artifact:
+          artifact_name: core-image-full-cmdline-20160330201408
```
Comment (pasinskim) [inline docs/devices_api.yml]: As we agreed to use human-readable name, maybe we can change this to `artifact_name: mender-1.1`?

File: docs/devices_api.yml (hunk 1 above)
Comment (bboozzoo) [inline docs/devices_api.yml]: done

File: docs/devices_api.yml (hunk 2 above)
Comment (bboozzoo) [inline docs/devices_api.yml]: changed do `my-app-0.1`

**Comment by bboozzoo** [comment]
Patch updated, please review & merge

**Pull request merged**